domain/topn_slow_query.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package domain
+
+import (
+	"container/heap"
+	"time"
+
+	"github.com/pingcap/tidb/util/execdetails"
+)
+
+type slowQueryHeap struct {
+	data []*slowQueryInfo
+}
+
+func (h *slowQueryHeap) Len() int           { return len(h.data) }
+func (h *slowQueryHeap) Less(i, j int) bool { return h.data[i].duration < h.data[j].duration }
```

Comment:
should it be `h.data[i].duration > h.data[j].duration`

Reply:
It's `Less` ... @zz-jason

Reply:
use heap[len(heap)-1] stores the slowest query?

Reply:
`heap[0]` is the fastest slow query, `heap[len(heap)-1]` may not be the slowest query.